ludwig/datasets/loaders/av_mnist.py

```diff
@@ -0,0 +1,214 @@
+# Copyright (c) 2022 Predibase, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import logging
+import os
+import re
+import struct
+from multiprocessing.pool import ThreadPool
+from typing import List, Optional
+
+import numpy as np
+import pandas as pd
+import torch
+from sklearn.decomposition import PCA
+
+from ludwig.datasets.dataset_config import DatasetConfig
+from ludwig.datasets.loaders.dataset_loader import DatasetLoader
+from ludwig.utils.fs_utils import makedirs
+
+logger = logging.getLogger(__name__)
+NUM_LABELS = 10
+
+
+class AV_MNISTLoader(DatasetLoader):
+    def __init__(
+        self, config: DatasetConfig, cache_dir: Optional[str] = None, transform=None, modal_separate=None, modal=None
+    ):
+        try:
+            from torchvision.io import write_png
+
+            self.write_png = write_png
+            self.transform = transform
+            self.modal_separate = modal_separate
+            self.modal = modal
+            self.audio_data = None
+            self.mnist_data = None
+            self.labels = None
+            self.data = None
+            self.file_names = {
+                "train_data": "train-images-idx3-ubyte.gz",
+                "train_labels": "train-labels-idx1-ubyte.gz",
+                "test_data": "t10k-images-idx3-ubyte.gz",
+                "test_labels": "t10k-labels-idx1-ubyte.gz",
```

Comment:
Should we grab these from the `config` object so that if there are any changes in the files names within the config, the loader will also use the updated names?

Reply:
Done